src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -55,6 +55,18 @@
 // Disabling pending crashes
 #define ENABLE_CONNECTIVITY_MONITORING 0
 
+#ifdef DEBUG
+#define MTR_REQUIRE_SUBCLASS_IMPL_PASTED(message) \
+    MTR_LOG_ERROR(message);                       \
+    NSAssert(NO, @message)
+#else // DEBUG
+#debug MTR_REQUIRE_SUBCLASS_IMPL_PASTED(message) \
+    MTR_LOG_ERROR(message)
+#endif // DEBUG
+
+#define MTR_REQUIRE_SUBCLASS_IMPL(selectorString) \
+    MTR_REQUIRE_SUBCLASS_IMPL_PASTED("MTRDevice " selectorString " must be handled by subclasses")
```

Comment:
Just use `NSStringFromSelector(_cmd)`

Reply:
That returns an NSString, so to concatenate it with stuff I'll have to do some more complicated bits, right?

Reply:
yeah, `MTR_LOG_ERROR` is just a wrapper around the C-format-string-wanting `ChipLogError` - but, it does take variadic parameters, so you could use `%s` with `[NSStringFromSelector(_cmd) cStringUsingEncoding:NSUTF8StringEncoding]`

Reply:
MTR_LOG_ERROR / ChiplogError supports '%@' placeholders on Darwin, you don't need to do anything special.

Reply:
And, and I can use formatting with NSAssert, too.  OK, great.